### PR TITLE
05 - Added two events

### PR DESCRIPTION
**Pull request by xavierlepretre**

### CODE REVIEW

File: x/checkers/keeper/msg_server_play_move.go [hunk 1]
```diff
@@ -52,6 +53,17 @@ func (k msgServer) PlayMove(goCtx context.Context, msg *types.MsgPlayMove) (*typ
...
+	// What to emit
+	ctx.EventManager().EmitEvent(
+		sdk.NewEvent(types.PlayMoveEventKey,
+			sdk.NewAttribute(types.PlayMoveEventCreator, msg.Creator),
+			sdk.NewAttribute(types.PlayMoveEventIdValue, msg.IdValue),
+			sdk.NewAttribute(types.PlayMoveEventCapturedX, strconv.FormatInt(int64(captured.X), 10)),
+			sdk.NewAttribute(types.PlayMoveEventCapturedY, strconv.FormatInt(int64(captured.Y), 10)),
+			sdk.NewAttribute(types.PlayMoveEventWinner, fullGame.Game.Winner().Color),
+		),
+	)
```
Comment (marbar3778) [inline x/checkers/keeper/msg_server_play_move.go:65]:
```suggestion
	ctx.EventManager().EmitEvent(
		sdk.NewEvent(sdk.EventTypeMessage,
		sdk.NewAttribute(sdk.AttributeKeyModule, types.AttributeValueCategory),
		sdk.NewAttribute(sdk.AttributeKeyAction, types.PlayMoveEventKey),
			sdk.NewAttribute(types.PlayMoveEventCreator, msg.Creator),
			sdk.NewAttribute(types.PlayMoveEventIdValue, msg.IdValue),
			sdk.NewAttribute(types.PlayMoveEventCapturedX, strconv.FormatInt(int64(captured.X), 10)),
			sdk.NewAttribute(types.PlayMoveEventCapturedY, strconv.FormatInt(int64(captured.Y), 10)),
			sdk.NewAttribute(types.PlayMoveEventWinner, fullGame.Game.Winner().Color),
		),
	)
```

this is more in tune with sdk conventions.

File: x/checkers/keeper/msg_server_create_game.go [hunk 2]
```diff
@@ -28,5 +29,17 @@ func (k msgServer) CreateGame(goCtx context.Context, msg *types.MsgCreateGame) (
 	nextGame.IdValue++
 	k.Keeper.SetNextGame(ctx, nextGame)
 
-	return &types.MsgCreateGameResponse{}, nil
+	// What to emit
+	ctx.EventManager().EmitEvent(
+		sdk.NewEvent(types.StoredGameEventKey,
```
Comment (marbar3778) [inline x/checkers/keeper/msg_server_create_game.go:34]: see below message to brings event into sdk standard

File: x/checkers/keeper/msg_server_play_move.go (hunk 1 above)
Comment (xavierlepretre) [reply x/checkers/keeper/msg_server_play_move.go:65]: What do you have in mind for the content of `types.AttributeValueCategory`?

Seeing what is declared after `sdk.EventTypeMessage` I wonder if I should also add:

```go
sdk.NewAttribute(sdk.AttributeKeyModule, "checkers")
```

Comment (marbar3778) [reply x/checkers/keeper/msg_server_play_move.go:65]: you could use the module name. not sure if you have it defined in the types folder. This is easier than using a string.

Comment (xavierlepretre) [reply x/checkers/keeper/msg_server_play_move.go:65]: And what about `types.AttributeValueCategory`?

Comment (marbar3778) [reply x/checkers/keeper/msg_server_play_move.go]: in sdk based modules AttributeValueCategory is an alias of moduleName

**Review by marbar3778 — APPROVED** [review]

**Pull request closed without merging**